### PR TITLE
Add minStack.py

### PR DESCRIPTION
**Pull request by peinanteng**

### CODE REVIEW

File: minStack.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    def push(self, x):
+        """
+        :type x: int
+        :rtype: void
+        """
+        self.stack.append(x)
+        if len(self.minstack) == 0 or x <= self.minstack[-1]:
+            self.minstack.append(x)
+
```
Comment (nkukarl) [inline minStack.py]: One line spacing only.

File: minStack.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        if self.stack[-1] == self.minstack[-1]:
+            self.minstack.pop()
+        return self.stack.pop()
+
+    def top(self):
+        """
+        :rtype: int
+        """
+        return self.stack[-1]
+
```
Comment (nkukarl) [inline minStack.py]: One line spacing only.

File: minStack.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        self.stack.append(x)
+        if len(self.minstack) == 0 or x <= self.minstack[-1]:
+            self.minstack.append(x)
+        
+
+    def pop(self):
+        """
+        :rtype: void
+        """
+        if self.stack[-1] == self.minstack[-1]:
```
Comment (nkukarl) [inline minStack.py]: Shouldn't the last item of `minStack` be popped out always?
Why there is a condition here

File: minStack.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+class MinStack(object):
+
+    def __init__(self):
+        """
+        initialize your data structure here.
+        """
+        self.stack = []
+        self.minstack = []
```
Comment (nkukarl) [inline minStack.py]: `minStack`

File: minStack.py (hunk 1 above)
Comment (peinanteng) [reply minStack.py]: Sure

File: minStack.py (hunk 2 above)
Comment (peinanteng) [reply minStack.py]: Sure

File: minStack.py (hunk 3 above)
Comment (peinanteng) [reply minStack.py]: last item of minStack is popped out only if it equals to last item of stack

File: minStack.py (hunk 4 above)
Comment (peinanteng) [reply minStack.py]: sure

**Pull request merged**